### PR TITLE
Deployment preparation: `L2GaugeCheckpointer`

### PR DESCRIPTION
**Pull request by jubeira**
# Deployment preparation: `L2GaugeCheckpointer`

Build-info: monorepo-master@https://github.com/balancer/balancer-v2-monorepo/commit/cbce7d63479dafb4f4ea9ad8cb2dbdbb26edae50.

The block number is right before the [latest available manual checkpoint](https://etherscan.io/tx/0x617b441cac07386a37513dfdf351821793d795b3beb1aab1d71dad1bc69a7c86).

At that point we already have the `GaugeAdder` V4 deployed as well as its coordinator script. The only pending step was to run the coordinator.

## Deployment preparation tasks checklist:

- [x] Deployments `README` is updated with the new task name and link
- [x] Task `README` has a description and links to the artifacts
- [x] `build-info` is updated with artifacts compiled at the specified commit
- [x] Artifacts are generated from `build-info`
- [x] New task has a working fork test 

## Deprecating old tasks checklist: N/A

Note: existing `20221205-l2-gauge-checkpointer` task was removed since it was never used. See https://etherscan.io/address/0xf23b4DB826DbA14c0e857029dfF076b1c0264843 (no events emitted, no gauges added, etc). No permissions have been granted to the contract either.

## Code checklist:

- [x] The diff is legible and has no extraneous changes 
- [x] Complex code has been commented in tests and scripts
- [x] Fork test block number and test cases are appropriate for this deployment
- [x] The base branch is either `master`, or there's a description of how to merge

## Issue Resolution

N/A

### CODE REVIEW

File: tasks/20230527-l2-gauge-checkpointer/test/task.fork.ts [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+
+    const adaptorEntrypointTask = new Task(
+      '20221124-authorizer-adaptor-entrypoint',
+      TaskMode.READ_ONLY,
+      getForkedNetwork(hre)
+    );
+    adaptorEntrypoint = await adaptorEntrypointTask.deployedInstance('AuthorizerAdaptorEntrypoint');
+  });
+
+  // At this block, the adder ocordinator has been deployed but not executed.
```
Comment (EndymionJkb) [inline tasks/20230527-l2-gauge-checkpointer/test/task.fork.ts:126]:
```suggestion
  // At this block, the adder coordinator has been deployed but not executed.
```

File: tasks/20230527-l2-gauge-checkpointer/test/task.fork.ts [hunk 2]
```diff
@@ -0,0 +1,461 @@
...
+      Array.from(gauges).map(([gaugeType, gaugesData]) => {
+        L2GaugeCheckpointer.connect(admin).addGaugesWithVerifiedType(
+          GaugeType[gaugeType],
+          gaugesData.map((gaugeData) => gaugeData.address)
+        );
+      })
+    );
+  });
+
+  before('grant checkpoint permission to fees gauge checkpointer', async () => {
```
Comment (EndymionJkb) [inline tasks/20230527-l2-gauge-checkpointer/test/task.fork.ts:197]: Is this title right? "fees gauge checkpointer"?

**Review by EndymionJkb — DISMISSED** [review]
I verified that the build-info matches the specified commit in v2. Just a couple nits otherwise.

Very comprehensive and detailed fork test.

File: tasks/20230527-l2-gauge-checkpointer/test/task.fork.ts (hunk 2 above)
Comment (jubeira) [reply tasks/20230527-l2-gauge-checkpointer/test/task.fork.ts]: :grimacing: good catch.

**Pull request merged**